libtransmission/log.cc

```diff
@@ -247,12 +247,10 @@ void tr_logAddMessage(char const* file, int line, tr_log_level level, std::strin
 {
     TR_ASSERT(!std::empty(msg));
 
-    auto name_fallback = std::string{};
     if (std::empty(name))
     {
         auto const base = tr_sys_path_basename(file);
-        name_fallback = fmt::format(FMT_STRING("{}:{}"), !std::empty(base) ? base : "?", line);
-        name = name_fallback;
+        name = fmt::format(FMT_STRING("{}:{}"), !std::empty(base) ? base : "?", line);
```

Comment:
Ahh, the `log.cc` changes need to be reverted too. I didn't see this regression until the asan CI tests started failing:

`name` is `std::string_view` that is passed in by the user, and here we're providing a fallback if `name` is empty.

`name_fallback` needs to exist in the outer scope to keep alive the `std::string` that `name` _points to_ iff the caller doesn't provide a name.

Reply:
Would this work by any chance?
```
auto name_fallback = fmt::format(FMT_STRING("{}:{}"), !std::empty(base) ? base : "?", line);
name = name_fallback;
```

(just trying to see if we can avoid `= std::string{};`)

Reply:
@Coeur  the codebase tends towards [AAA style](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/) `auto name_fallback = std::string{};` instead of `std::string name_fallback;` in new code. If you look at the two in godbolt ([1](https://godbolt.org/z/rhsjoTv3f), [2](https://godbolt.org/z/3edGzYGn9)) the two are equivalent; there's no extra efficiency in avoiding `= std::string{}`.

`auto name_fallback = fmt::format...` is a little less efficient since we don't need to actually populate the string when `base` is nonempty.

Reply:
It's not for efficiency after compiler optimization, it's to avoid a warning regarding unused initial value.
Hence why `std::string name_fallback;` is possibly better than that "AAA style".

Reply:
I wonder if that warning is valid? Just constructing `std::string name_fallback;` will run its constructor so that the variable is initialized.

Can you paste the warning here?

Reply:
Maybe I'm wrong. @GermanAizek for telling us what was the warning that motivated this change?

Reply:
I reverted the log.cc change in 6381eb9